bindata/network/ovn-kubernetes/ovnkube-master.yaml

```diff
@@ -166,6 +166,22 @@ spec:
                   fi
                   sleep 2
                   done
+
+                  # Upgrade the db if required.
+                  DB_SCHEMA="/usr/share/ovn/ovn-nb.ovsschema"
+                  DB_SERVER="unix:/var/run/ovn/ovnnb_db.sock"
+                  schema_name=$(ovsdb-tool schema-name $DB_SCHEMA)
+                  db_version=$(ovsdb-client -t 10 get-schema-version "$DB_SERVER" "$schema_name")
+                  target_version=$(ovsdb-tool schema-version "$DB_SCHEMA")
+
+                  if ovsdb-tool compare-versions "$db_version" == "$target_version"; then
+                    :
+                  elif ovsdb-tool compare-versions "$db_version" ">" "$target_version"; then
+                      echo "Database $schema_name has newer schema version ($db_version) than our local schema ($target_version), possibly an upgrade is partially complete?"
+                  else
+                      echo "Upgrading database $schema_name from schema version $db_version to $target_version"
+                      ovsdb-client -t 30 convert "$DB_SERVER" "$DB_SCHEMA"
```

Comment:
Will the timeout be sufficient for large db sizes? Or is the conversion orthogonal to the size?
Should we check for the return value of `convert` to make sure the upgrade was successful? What is our recourse if the conversion fails or causes data loss?

Reply:
I copied much of the code from here - https://github.com/openvswitch/ovs/blob/master/utilities/ovs-lib.in#L461

ovn-ctl scripts call upgrade_cluster() which does the samething.

But since the  daemonset here calls, run_xx_ovsdb, it  never calls upgrade_cluster().

I'm not sure about the what should be proper timeout value. I'm fine if we want to increase the timeout value to may be 60 seconds ? or higher.

If convert fails, we can probably attempt few times before we give up.
I'm not sure what to do in case uprade fails or there seems to be data loss. 

@dceara  Do you have any comments here ?

Reply:
I discussed with @dceara.

We both are of the opinion that - if the upgrade fails, the only thing we can do is delete all DBs and start over.